### ISSUE TITLE
False positive assignment-from-no-return for collections.deque.popleft() on Python 3.7

### ISSUE THREAD

**Issue by dgilland** [PyCQA/pylint#2267]
### Steps to reproduce
1. Create test file with the following:

```python
# test.py
from collections import deque

def test():
    queue = deque([1, 2, 3])
    item = queue.popleft()
    return item

print(test())
```

2. Run `pylint test.py -E`

### Current behavior
```
$ pylint test.py -E
************* Module test
test.py:5:4: E1111: Assigning to function call which doesn't return (assignment-from-no-return)
```

### Expected behavior
No error should be given since `deque.popleft()` returns a value.

### pylint --version output
```
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.7.0 (default, Jul  6 2018, 19:08:39)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-18)]
```

**Comment by cburkhardtBB** [comment]
This issue still exists.

**Comment by PCManticore** [comment]
This issue was fixed in the new release, with what version do you see it?

**Comment by cburkhardtBB** [comment]
pylint 2.0
python 3.6
Specifically when using a Python property decorator to return the value instead of a return statement inside the function. This error is thrown even though the @dynamo_write_handler is returning a False or True bool value

ERROR: E1111: Assigning to function call which doesn't return (assignment-from-no-return)

 @dynamo_write_handler
    def update_session(self, session_model: VideoSessionModel) -> bool:
        """
        :param session_model: VideoSessionModel
        :param event_detail: EventDetailModel
        :return: Dynamo write handler returns True if concurrency failure, False if successful
        """
        update_model = session_model.to_primitive()
        query = [
            'version=:nv',
            'last_timestamp=:lts',
            '#state=:st',
            'program_type=:pt'
        ]
        attribute_values = {
            ':cv': update_model['version'],
            ':nv': update_model['version'] + 1,
            ':lts': update_model['last_timestamp']
        }
        attribute_keys = {
            '#state': 'state'
        }

        self.table.update_item(
            Key={
                'profile_id': update_model['profile_id'],
                'session_range_id': update_model['session_range_id']
            },
            UpdateExpression='SET ' + ', '.join(query),
            ConditionExpression='version=:cv',
            ExpressionAttributeValues=attribute_values,
            ExpressionAttributeNames=attribute_keys,
            ReturnValues="NONE"
        )
